yamllint/rules/anchors.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2021 Sergei Mikhailov
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Use this rule to prevent aliases pointing to pointers that don't exist.
+
+.. rubric:: Examples
+
+#. With ``anchors: {}``
+
+   the following code snippet would **PASS**:
+   ::
+
+    address: &address |
+      Williams St. 13
+    target_address: *address
+
+   the following code snippet would **FAIL**:
+   ::
+
+    address: &address |
+      Williams St. 13
+    target_address: *wrong_address
+
+#. This appends to merges as well:
+
+   the following code snippet would **PASS**:
+   ::
+
+    default_address: &address
+      state: South Carolina
+      city:  Barnwell
+    target_address:
+      <<: *address
+      city: Barnaul
+
+   the following code snippet would **FAIL**:
+   ::
+
+    default_address: &address
+      state: South Carolina
+      city:  Barnwell
+    target_address:
+      <<: *wrong_address
+      city: Barnaul
+
+#. Duplicate anchors are not allowed.
+
+   the following code snippet would **PASS**:
+   ::
+
+    clients:
+        jack: &jack_client
+            billing_id: 1234
+        bill: &bill_client
+            billing_id: 5678
+    target_client: *jack_client
+
+   the following code snippet would **FAIL**:
+   ::
+
+    clients:
+        jack: &jack_client
+            billing_id: 1234
+        bill: &jack_client
+            billing_id: 5678
+    target_client: *jack_client
+"""
+
+from yaml import DocumentStartToken, AnchorToken, AliasToken, BlockEndToken
+
+from yamllint.linter import LintProblem
+
+
+ID = 'anchors'
+TYPE = 'token'
+
+
+def check(__conf, token, __prev, __next, __nextnext, context):
```

Comment:
I guess these `__` are for unused variables. Other rules don't use that, so I prefer keeping the same code for all.

Reply:
I'd suggest to keep this syntax to highlight that variables is unused, and "declared" only in order to comply with the function args order.
It seems to be pretty intuitive, and also keeps linters\IDE from warning about unused variables.

Reply:
I don't know, why not but in this case it should be done for all yamllint, ideally enforced by `flake8`, and most of all done in a separated *refactor* PR. In the meantime, can you please remove them to make the code consistent? Maybe it's not obvious, but consistent code makes the PR / review process easier for everyone.